site/book/06-deploying-packages/00.md

```diff
@@ -0,0 +1,68 @@
+In the last chapter of this book, we are going to cover how you deploy a kpt
+package to a Kubernetes cluster and how the cluster state is managed as the
+package evolves over time.
+
+We use `kpt live apply` instead `kubectl apply` since it provides some critical
+functionality not provided by the latter: pruning and reconciled status. To
+enable this functionality, we need a cluster-side mechanism for grouping and
+tracking resources belonging to a package. This cluster-side grouping is
+implemented using a custom resource of kind `ResourceGroup`. Otherwise,
+`kpt live` and `kubectl` are complementary. For example, you can use
+`kubectl get` as normal would.
+
+## Pruning
+
+`live apply` will automatically delete cluster resources that are no longer
+present in the local package. This clean-up functionality is called pruning.
+
+For example, consider a package which has been applied with the following three
+resources:
+
+```
+service-1 (Service)
+deployment-1 (Deployment)
+config-map-1 (ConfigMap)
+```
+
+Then imagine the package is updated to contain the following resources:
+
+```
+service-1 (Service)
+deployment-1 (Deployment)
+config-map-2 (ConfigMap)
+```
+
+When the updated package is applied, `config-map-1` is automatically pruned from
+the cluster.
+
+## Reconcile Status
+
+Kubernetes is based on an asynchronous reconciliation model. When you apply a
+resource to a cluster, you actually care about two different things:
+
+- Did the resource apply successfully (synchronous)?
+- Did the resource reconcile successfully (asynchronous)?
+
+This is referred to as _apply status_ and _reconcile status_ respectively:
+
+![img](/static/images/status.svg)
+
+The `live apply` command computes the reconcile status. An example of this could
+be applying a `Deployment`. Without computing reconcile status, the operation
+would be reported as successful as soon as the resource has been created in the
+API server. With reconcile status, `live apply` will wait until the desired
+number of pods have been created and become available.
+
+For core kubernetes types, reconcile status is computed using hardcoded rules.
+For CRDs, the status computation is based on recommended [convention for status
+fields] that needs to be followed by custom resource publishers. If CRDs follow
+these conventions, `live apply` will be able to correctly compute status.
+
+Once you know the reconcile status of a single resource, then you really want to
```

Comment:
I'm not sure about this section. We used to have the concept of aggregate status in kpt live, but we eventually moved away from it. For `kpt live apply`, what we really care about isn't a single aggregate status, but that each resource reach its desired end state. If the package contains a combination of new resources and updates to existing resources, as well as lacking some resources that should be pruned, what we actually want to wait for is all new and updated resources to be reconciled and all pruned resources to be fully removed from the cluster (i.e. all finalizers have completed).